socrata-metadata/socrata_metadata_pub.py

```diff
@@ -0,0 +1,208 @@
+    central_timezone = pytz.timezone("US/Central")
+    central_datetime = utc_datetime.astimezone(central_timezone)
+
+    # To socrata format
+    central_timestamp_string = central_datetime.strftime("%Y-%m-%dT%H:%M:%S")
+
+    return central_timestamp_string
+
+
+def transform(data):
+    logger.info("Transforming data")
+    basic = HTTPBasicAuth(username=SO_USER, password=SO_PASS)
+
+    output_data = []
+    for row in data:
+        row["resource"]["page_views_last_week"] = row["resource"]["page_views"][
+            "page_views_last_week"
+        ]
+        row["resource"]["page_views_last_month"] = row["resource"]["page_views"][
+            "page_views_last_month"
+        ]
+        row["resource"]["page_views_total"] = row["resource"]["page_views"][
+            "page_views_total"
+        ]
+        filtered_data = {key: row["resource"][key] for key in OUT_FIELDS}
+
+        # URL to dataset is based on if it is public or not
+        if row["is_public"]:
+            filtered_data["dataset_url"] = BASE_URL + filtered_data["id"]
+            filtered_data["is_public"] = True
+        else:
+            filtered_data["dataset_url"] = EDP_URL + filtered_data["id"]
+            filtered_data["is_public"] = False
+
+        # Convert UTC time to local time
+        filtered_data["updatedAt"] = convert_tz(filtered_data["updatedAt"])
+        filtered_data["createdAt"] = convert_tz(filtered_data["createdAt"])
+        filtered_data["metadata_updated_at"] = convert_tz(
+            filtered_data["metadata_updated_at"]
+        )
+        filtered_data["data_updated_at"] = convert_tz(filtered_data["data_updated_at"])
+        filtered_data["publication_date"] = convert_tz(
+            filtered_data["publication_date"]
+        )
+
+        filtered_data["owner_display_name"] = row["owner"]["display_name"]
+
+        if "domain_tags" in row["classification"]:
+            filtered_data["tags"] = ", ".join(
+                str(x) for x in row["classification"]["domain_tags"]
+            )
+        else:
+            filtered_data["tags"] = ""
+
+        metadata = {}
+        if "domain_private_metadata" in row["classification"]:
+            for item in row["classification"]["domain_private_metadata"]:
+                metadata[item["key"]] = item["value"]
+        for item in row["classification"]["domain_metadata"]:
+            metadata[item["key"]] = item["value"]
+
+        fields = {
+            "Publishing-Information_Automation-Method": "automation_method",
+            "Publishing-Information_Automation-Method-(if-Other)": "automation_method_other",
+            "Publishing-Information_Update-Frequency": "update_frequency",
+            "Publishing-Information_Spatial-Information": "spatial_information",
+            "Ownership_Department-name": "department_name",
+            "Ownership_Program-Name": "program_name",
+            "Strategic-Area_Strategic-Direction-Outcome": "strategic_area",
+        }
```

Comment:
would be good to pull this out of the main loop and declare the dict one time only